cpp/tests/community/k_truss_test.cu

```diff
@@ -0,0 +1,408 @@
+  {
+    constexpr bool renumber = false;
+
+    auto [k_truss_usecase, input_usecase] = param;
+
+    raft::handle_t handle{};
+
+    HighResTimer hr_timer{};
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.start("MG Construct graph");
+    }
+
+    // NX k_truss is not implemented for graph with self loop and isolated
+    // vertices therefore dropped them especially for rmat generated graphs
+    auto [graph, edge_weights, d_renumber_map_labels] =
+      cugraph::test::construct_graph<vertex_t, edge_t, weight_t, false, false>(
+        handle, input_usecase, k_truss_usecase.test_weighted_, renumber, true, true);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    auto graph_view = graph.view();
+
+    auto edge_weight_view =
+      edge_weights ? std::make_optional((*edge_weights).view()) : std::nullopt;
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.start("K-truss");
+    }
+
+    auto [d_cugraph_src, d_cugraph_dst, d_cugraph_wgt] =
+      cugraph::k_truss<vertex_t, edge_t, weight_t, false>(
+        handle, graph_view, edge_weight_view, k_truss_usecase.k, false);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    if (k_truss_usecase.check_correctness) {
+      auto [d_reference_src, d_reference_dst, d_reference_wgt] =
+        k_truss_reference<vertex_t, edge_t, weight_t>(
+          handle, graph_view, edge_weight_view, k_truss_usecase.k);
+      if (edge_weight_view) {
+        auto d_edge_first = thrust::make_zip_iterator(thrust::make_tuple(
+          d_cugraph_src.begin(), d_cugraph_dst.begin(), (*d_cugraph_wgt).begin()));
+
+        thrust::sort(handle.get_thrust_policy(), d_edge_first, d_edge_first + d_cugraph_src.size());
+      } else {
+        auto d_edge_first = thrust::make_zip_iterator(
+          thrust::make_tuple(d_cugraph_src.begin(), d_cugraph_dst.begin()));
+
+        thrust::sort(handle.get_thrust_policy(), d_edge_first, d_edge_first + d_cugraph_src.size());
+      }
+
+      EXPECT_EQ(d_cugraph_src.size(), d_reference_src.size());
+
+      ASSERT_TRUE(thrust::equal(handle.get_thrust_policy(),
+                                d_cugraph_src.begin(),
+                                d_cugraph_src.end(),
+                                d_reference_src.begin()));
+
+      ASSERT_TRUE(thrust::equal(handle.get_thrust_policy(),
+                                d_cugraph_dst.begin(),
+                                d_cugraph_dst.end(),
+                                d_reference_dst.begin()));
```

Comment:
You can bring cugraph results using to_host and use std::equal (so, this file can be .cpp, we expect our users to call algorithms from .cpp files not requiring CUDA compiler).

Reply:
This file is now `.cpp` and the comparison is performed on std vectors through `std::equal`.